src/wallet/wallet.cpp

```diff
@@ -1818,32 +1818,37 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
     return result;
 }
 
+std::vector<CWalletTx*> CWallet::GetSortedTxs()
+{
+    AssertLockHeld(cs_wallet);
+    // Sort pending transactions based on their initial wallet insertion order
+    std::map<int64_t, CWalletTx*> sorted_txs;
+    for (auto& [wtxid, wtx] : mapWallet) {
+        assert(wtx.GetHash() == wtxid);
+        sorted_txs.insert(std::make_pair(wtx.nOrderPos, &wtx));
+    }
+    std::vector<CWalletTx*> out;
+    for (auto [pos, wtx] : sorted_txs) {
```

Comment:
nit: not sure if we need `pos` here doesn't seem to be used

Reply:
Done